using_restricted_array.rb

```diff
@@ -1,165 +1,234 @@
-end
-puts
-
-# print the largest value in the array
-largest = find_largest(another_array, another_array_length)
-puts "The largest value in the array is #{largest}"
-puts
-
-# sort the array
-sort(another_array, another_array_length)
-print "Array sorted in ascending order: "
-print_array(another_array)
-puts
-
-# delete the first entry with the value_to_delete
-value_to_delete = another_array[another_array_length/2]
-delete(another_array, another_array_length, value_to_delete)
-print "#{value_to_delete} deleted from array: "
-print_array(another_array)
-puts
-
-# delete the first entry with the value_to_delete
-value_to_delete = another_array[another_array_length/2]
-delete(another_array, another_array_length, value_to_delete)
-print "#{value_to_delete} deleted from array: "
-print_array(another_array)
-puts
-
-# print the largest value in the array
-largest = find_largest(another_array, another_array_length)
-puts "The largest value in the array is #{largest}"
-puts
-
-# sort the array
-sort(another_array, another_array_length)
-print "Array sorted in ascending order: "
-print_array(another_array)
-puts
-
-# insert 123 in to the array sorted in ascending order
-value_to_insert = 123
-insert_ascending(another_array, another_array_length, value_to_insert)
-print "#{value_to_insert} inserted into the array: "
-print_array(another_array)
-puts
-
-# empty array
-empty(another_array, another_array_length)
-print "Emptied array looks like: "
-print_array(another_array)
-puts
-
-# insert 123 in to the array sorted in ascending order
-value_to_insert = 123
-insert_ascending(another_array, another_array_length, value_to_insert)
-print "#{value_to_insert} inserted into the array: "
-print_array(another_array)
-puts
+require_relative 'restricted_array.rb'
+# RestrictedArray can be created using a specified size, or a random size in
+# the range of 1-20 will be chosen for you.
+# All values are integers in the range of 1-221.
+# RestrictedArray cannot be resized.
+# An unused space in the restricted array has the 'SPECIAL_VALUE'
+SPECIAL_VALUE = 9999
+
+## Calculates the length of the restricted integer array_size
+def length(array)
+    counter = 0
+
+    counter += 1 until array[counter].nil?
+
+    counter
+end
+
+# Prints each integer values in the array
+def print_array(array)
+    i = 0
+    array_length = length(array)
+
+    array_length.times do
+        print (array[i]).to_s
+        i += 1
+    end
+end
+
+# Reverses the values in the integer array
+def reverse(array, length) # Ruby
+    return if length < 2
+
+    i = 0
+    j = length - 1
+
+    until j < i
+        num_holder = array[i]
+        array[i] = array[j]
+        array[j] = num_holder
+
+        i += 1
+        j -= 1
+    end
+
+    array
+end
+
+# For an unsorted array, searches for 'value_to_find'.
+# Returns true if found, false otherwise.
+def search(array, length, value_to_find)
+    (0...length).each do |i|
+        return true if array[i] = value_to_find
```

Comment:
This should be comparison instead of assignment. i.e. `if array[i] == value_to_find` and not `if array[i] = value_to_find`, which will always be true.